interfaces/connection.go

```diff
@@ -111,6 +111,16 @@ func NewConnectedSlot(slot *snap.SlotInfo, dynamicAttrs map[string]interface{})
 	}
 }
 
+// NewConnectedSlotReload creates an object representing a connected slot that is reloaded with static attributes
+// restored from state (not from the SlotInfo).
+func NewConnectedSlotReload(slot *snap.SlotInfo, staticAttrs, dynamicAttrs map[string]interface{}) *ConnectedSlot {
+	return &ConnectedSlot{
+		slotInfo:     slot,
+		staticAttrs:  copyAttributes(staticAttrs),
+		dynamicAttrs: utils.NormalizeInterfaceAttributes(dynamicAttrs).(map[string]interface{}),
```

Comment:
the normalize is just for extra safety right?  we store things already normalized in conns?

Reply:
Indeed.